### PR TITLE
Implement runtime-extensible syntax highlighting

### LINKED ISSUE #30: Add syntax highlighting
As stated in the README, syntax highlighting is pretty straightforward to implement, since the AST is already parsed into text tokens and whitespace.  So we just need to assign each text token to a syntax category when rendering, and then we get syntax highlighting.

### PR DESCRIPTION
**Pull request by kneasle**
Fixes #30.  I doubt many editors get accurate syntax highlighting within about two months...

### CODE REVIEW

**Comment by kneasle** [comment]
@Kixunil, it seems like I can't request a review conventionally :shrug: but could you pls look at my code and see what you think :grin:.  Cheers!

File: src/ast/json.rs [hunk 1]
```diff
@@ -80,24 +80,33 @@ impl<'arena> Ast<'arena> for JSON<'arena> {
     ) -> Vec<RecTok<'arena, Self>> {
         let is_pretty = format_style == &JSONFormat::Pretty;
         match self {
-            JSON::True => vec![RecTok::Tok(DisplayToken::Text("true".to_string()))],
-            JSON::False => vec![RecTok::Tok(DisplayToken::Text("false".to_string()))],
-            JSON::Null => vec![RecTok::Tok(DisplayToken::Text("null".to_string()))],
-            JSON::Str(string) => vec![RecTok::Tok(DisplayToken::Text(format!(r#""{}""#, string)))],
+            JSON::True => vec![RecTok::Tok(DisplayToken::Text("true".to_string(), "const"))],
+            JSON::False => vec![RecTok::Tok(DisplayToken::Text(
+                "false".to_string(),
```
Comment (Kixunil) [inline src/ast/json.rs:85]: I believe `.to_owned()` is better than `.to_string()`

* Better conveys what's the point (`"..."` is already a string)
* Doesn't rely on specialization to be performant

However, I wonder if it's feasible to make it not allocate at all (by using `Cow<'static, str>`). IDK the specifics of `RecTok` or `DisplayToken` and don't have the time to study it right now.

File: src/editor/mod.rs [hunk 2]
```diff
@@ -100,13 +105,18 @@ impl<'arena, Node: Ast<'arena> + 'arena> Editor<'arena, Node> {
...
+                        // Hash the ref to decide on the colour
                         let mut hasher = DefaultHasher::new();
                         node.hash(&mut hasher);
                         let hash = hasher.finish();
                         cols[hash as usize % cols.len()]
+                    } else {
+                        *self
+                            .color_scheme
+                            .get(category)
+                            .unwrap_or(&Color::LIGHT_MAGENTA)
```
Comment (Kixunil) [inline src/editor/mod.rs:119]: In the future we may want to log unknown categories and ideally make sure each is logged only once. It'll make it easy to discover missing configuration or typos.

**Review by Kixunil — APPROVED** [review]

File: src/ast/json.rs (hunk 1 above)
Comment (kneasle) [reply src/ast/json.rs:85]: We aren't going to be modifying `DisplayToken`s after they're returned, so I'm thinking we have `DisplayToken::Text(String, SyntaxCategory)` and `DisplayToken::TextStr(&'static str, SyntaxCategory)`, which would stop us from having to spam `Cow::from` everywhere.

File: src/editor/mod.rs (hunk 2 above)
Comment (kneasle) [reply src/editor/mod.rs:119]: Oh good point - I'll fix that now.

File: src/ast/json.rs (hunk 1 above)
Comment (kneasle) [reply src/ast/json.rs:85]: Ok it's been `Cow`ified.

**Pull request merged**